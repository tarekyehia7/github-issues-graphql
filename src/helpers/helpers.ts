import formatDistanceToNowStrict from 'date-fns/formatDistanceToNowStrict';

export const getPagesNumber = (totalIssuesCount: number, issuesPerPage: number): number => {
	const totalPages = Math.ceil(totalIssuesCount / issuesPerPage);
	return !totalPages ? 1 : totalPages;
};

export const formatDate = (date: string) => {
	try {
		const myDate = new Date(date);
		const myFormattedDate = formatDistanceToNowStrict(myDate, { addSuffix: true });
		return myFormattedDate;
	} catch (e) {
		console.error(e);
		return '';
	}
};
