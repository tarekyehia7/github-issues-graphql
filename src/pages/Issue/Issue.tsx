import React from "react";
import { useNavigate, useParams } from "react-router-dom";

import { useGetIssueQuery } from "../../graphql/generatedTypes/graphql";
import { IssueDetailsComment } from "../../components/issueDetailsComment/IssueDetailsComment";
import { IssueDetailsHeader } from "../../components/issueDetailsHeader/IssueDetailsHeader";

export const Issue = () => {
	const { issueId } = useParams();
	const navigate = useNavigate();
	const { data, error, loading } = useGetIssueQuery({
		variables: {
			issueId: issueId ? parseInt(issueId) : 0,
		},
	});

	const issueData = data?.repository?.issue;
	const issueEdges = data?.repository?.issue?.comments?.edges;

	if (error) navigate("/");
	if (loading) return null;

	return (
		<>
			{issueData && issueData.author && (
				<>
					<IssueDetailsHeader
						title={issueData.title}
						state={issueData.state}
						issueId={issueId ?? ""}
						authorName={issueData.author.login}
						createdAt={issueData.createdAt}
					/>
					<IssueDetailsComment
						authorUrl={issueData.author.url}
						avatarUrl={issueData.author.avatarUrl}
						authorName={issueData.author.login}
						createdAt={issueData.createdAt}
						bodyHTML={issueData.bodyHTML}
					/>
				</>
			)}
			{issueEdges &&
				issueEdges.map((edge, idx) => {
					const comment = edge?.node;
					const author = comment?.author;
					if (!comment || !author) return null;
					return (
						<IssueDetailsComment
							key={`issue-comment-${idx}`}
							authorUrl={author.url}
							avatarUrl={author.avatarUrl}
							authorName={author.login}
							createdAt={comment.issue.createdAt}
							bodyHTML={comment.issue.bodyHTML}
						/>
					);
				})}
		</>
	);
};
