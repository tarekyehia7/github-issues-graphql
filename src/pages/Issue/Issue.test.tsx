import React from 'react';
import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MockedProvider } from '@apollo/client/testing';

import { IssuePage } from './Issue';
import { issueGraphQlMock } from '../../graphql/queries/mocks/issue.mock';
import { PageWithTheme } from '../../helpers/testing/helpers';
import { formatDate } from '../../helpers/helpers';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useParams: () => ({
		issueId: '25264',
	}),
}));

const repositoryIssue = issueGraphQlMock.result.data.repository.issue;
const repositoryEdges = issueGraphQlMock.result.data.repository.issue.comments.edges;

const renderPage = () => {
	const {
        container,
        getByText,
        getAllByText,
        getAllByTestId,
        getAllByAltText,
    } = render(
		<PageWithTheme>
            <MockedProvider mocks={[issueGraphQlMock]} addTypename={true}>
                <IssuePage />
            </MockedProvider>
        </PageWithTheme>,
	);

    return {
        container,
        getByText,
        getAllByText,
        getAllByTestId,
        getAllByAltText
    };
};

describe('<Issue />', () => {

	it('Should matches snapshot', async () => {
        const { container } = renderPage();

        await waitFor(() => new Promise((res) => setTimeout(res, 100)));
		expect(container).toMatchSnapshot();
	});

	it('should have avatar', async () => {
        const { getByText, getAllByAltText } = renderPage();
		await waitFor(() => {
			expect(getByText(repositoryIssue.title.trim())).toBeInTheDocument();

			expect(getAllByAltText('avatar url')[1]).toHaveAttribute(
				'src',
				repositoryEdges[0].node.author.avatarUrl,
			);
		});
	});

    it('Should have correct title - author comment', async () => {
        const { getAllByText } = renderPage();
        const cardTitle = `commented ${formatDate(repositoryIssue.createdAt)} ago`;

        await waitFor(() => {
			expect(getAllByText(cardTitle.trim())[0]).toBeInTheDocument();
		});
    });
    
    it('Should have correct number of cards', async () => {
        const { getAllByTestId } = renderPage();

        // increment by one because of first card is not represented as an edge
        const cardsLength = repositoryEdges.length + 1;

        await waitFor(() => {
            expect(getAllByTestId('card')).toHaveLength(cardsLength);
        });
    });

});
