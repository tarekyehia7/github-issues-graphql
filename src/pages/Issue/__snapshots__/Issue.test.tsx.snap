// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Issue /> Should matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ hNoBki"
  >
    <h1
      class="sc-hKMtZM"
    >
      <span>
        Bug: &lt;img/&gt; renders twice in firefox with react 18
      </span>
       #
      25264
    </h1>
    <div
      class="sc-gsnTZi cZkHAJ"
    >
      <span
        class="sc-jSMfEi pDLoo"
      >
        <svg
          aria-hidden="true"
          data-view-component="true"
          height="16"
          version="1.1"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
          />
          <path
            d="M8 0a8 8 0 100 16A8 8 0 008 0zM1.5 8a6.5 6.5 0 1113 0 6.5 6.5 0 01-13 0z"
          />
        </svg>
        open
      </span>
      kapibaara
       opened this issue 
      8 days
       ago
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/kapibaara"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/41840872?v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/kapibaara"
        >
          kapibaara
        </a>
        <span>
           commented 
          8 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        

        <p
          dir="auto"
        >
          React version: 18.2.0
        </p>
        

        <h2
          dir="auto"
        >
          Steps To Reproduce
        </h2>
        

        <ol
          dir="auto"
        >
          

          <li>
            Insert a tag  
            <code
              class="notranslate"
            >
              &lt;img/&gt;
            </code>
             with src
          </li>
          

          <li>
            Go to 
            <strong>
              Firefox
            </strong>
             inspect tool to the tab network
          </li>
          

          <li>
            Reload the page and watch image request in network
          </li>
          

        </ol>
        


        <p
          dir="auto"
        >
          Link to code example: 
          <a
            href="https://codesandbox.io/s/unruffled-jerry-9hli44?file=/src/App.js"
            rel="nofollow"
          >
            https://codesandbox.io/s/unruffled-jerry-9hli44?file=/src/App.js
          </a>
        </p>
        


        <h2
          dir="auto"
        >
          The current behavior
        </h2>
        

        <p
          dir="auto"
        >
          There are two request for image in firefox, image renders twice
          <br />
          

          <a
            href="https://user-images.githubusercontent.com/41840872/189360604-c1f1451a-be7f-497d-b9db-9d18f98af607.png"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/41840872/189360604-c1f1451a-be7f-497d-b9db-9d18f98af607.png"
              style="max-width: 100%;"
              width="1792"
            />
          </a>
        </p>
        

        <h2
          dir="auto"
        >
          The expected behavior
        </h2>
        

        <p
          dir="auto"
        >
          There is one request and one render in firefox
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/MiguelMachado-dev"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/29252011?u=ababd38859887c5e1854f1c48dfc94787ae91724&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/MiguelMachado-dev"
        >
          MiguelMachado-dev
        </a>
        <span>
           commented 
          7 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          It really happens, even on Chrome/Edge. But I think this is due to StrictMode. Could please try deleting StrictMode on 
          <code
            class="notranslate"
          >
            index.js
          </code>
          ?
        </p>
        

        <p
          dir="auto"
        >
          It will look like this.
        </p>
        

        <div
          class="highlight highlight-source-js notranslate position-relative overflow-auto"
          data-snippet-clipboard-copy-content="root.render(<App />);"
          dir="auto"
        >
          <pre>
            <span
              class="pl-s1"
            >
              root
            </span>
            <span
              class="pl-kos"
            >
              .
            </span>
            <span
              class="pl-en"
            >
              render
            </span>
            <span
              class="pl-kos"
            >
              (
            </span>
            <span
              class="pl-c1"
            >
              &lt;
            </span>
            <span
              class="pl-ent"
            >
              App
            </span>
             
            <span
              class="pl-c1"
            >
              /
            </span>
            <span
              class="pl-c1"
            >
              &gt;
            </span>
            <span
              class="pl-kos"
            >
              )
            </span>
            <span
              class="pl-kos"
            >
              ;
            </span>
          </pre>
        </div>
        

        <p
          dir="auto"
        >
          On StrictMode react render twice. 
          <a
            href="https://reactjs.org/docs/strict-mode.html"
            rel="nofollow"
          >
            https://reactjs.org/docs/strict-mode.html
          </a>
        </p>
        

        <blockquote>
          

          <p
            dir="auto"
          >
            StrictMode renders components twice (on dev but not production) in order to detect any problems with your code and warn you about them
          </p>
          

        </blockquote>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/kapibaara"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/41840872?v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/kapibaara"
        >
          kapibaara
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <blockquote>
          

          <p
            dir="auto"
          >
            It really happens, even on Chrome/Edge. But I think this is due to StrictMode. Could please try deleting StrictMode on 
            <code
              class="notranslate"
            >
              index.js
            </code>
            ?
          </p>
          

          <p
            dir="auto"
          >
            It will look like this.
          </p>
          

          <div
            class="highlight highlight-source-js notranslate position-relative overflow-auto"
            data-snippet-clipboard-copy-content="root.render(<App />);"
            dir="auto"
          >
            <pre>
              <span
                class="pl-s1"
              >
                root
              </span>
              <span
                class="pl-kos"
              >
                .
              </span>
              <span
                class="pl-en"
              >
                render
              </span>
              <span
                class="pl-kos"
              >
                (
              </span>
              <span
                class="pl-c1"
              >
                &lt;
              </span>
              <span
                class="pl-ent"
              >
                App
              </span>
               
              <span
                class="pl-c1"
              >
                /
              </span>
              <span
                class="pl-c1"
              >
                &gt;
              </span>
              <span
                class="pl-kos"
              >
                )
              </span>
              <span
                class="pl-kos"
              >
                ;
              </span>
            </pre>
          </div>
          

          <p
            dir="auto"
          >
            On StrictMode react render twice. 
            <a
              href="https://reactjs.org/docs/strict-mode.html"
              rel="nofollow"
            >
              https://reactjs.org/docs/strict-mode.html
            </a>
          </p>
          

          <blockquote>
            

            <p
              dir="auto"
            >
              StrictMode renders components twice (on dev but not production) in order to detect any problems with your code and warn you about them
            </p>
            

          </blockquote>
          

        </blockquote>
        

        <p
          dir="auto"
        >
          it doesn't help, in my example in codesandbox I tried to remove StrictMode, but the behavior is the same, there are two requests in network tab
          <br />
          
Аnd on Chrome, if you open this 
          <a
            href="https://c1i5n8.csb.app/"
            rel="nofollow"
          >
            https://c1i5n8.csb.app/
          </a>
           there are no two requests in network tab
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/MiguelMachado-dev"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/29252011?u=ababd38859887c5e1854f1c48dfc94787ae91724&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/MiguelMachado-dev"
        >
          MiguelMachado-dev
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          There is, check 
          <a
            href="https://codesandbox.io/s/unruffled-jerry-9hli44?file=/src/index.js"
            rel="nofollow"
          >
            https://codesandbox.io/s/unruffled-jerry-9hli44?file=/src/index.js
          </a>
          <br />
          
On Edge and Firefox I still getting 2 requests, disabling strict-mode it stops.
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/sigorilla"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/1208639?u=c531dee3f3f7cdb9657a78c54bbea4081f327158&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/sigorilla"
        >
          sigorilla
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          <a
            class="user-mention notranslate"
            data-hovercard-type="user"
            data-hovercard-url="/users/MiguelMachado-dev/hovercard"
            data-octo-click="hovercard-link-click"
            data-octo-dimensions="link_type:self"
            href="https://github.com/MiguelMachado-dev"
          >
            @MiguelMachado-dev
          </a>
          , Just check this example (without 
          <code
            class="notranslate"
          >
            StrictMode
          </code>
          ): 
          <a
            href="https://codesandbox.io/s/goofy-stonebraker-pf0wg7?file=/src/index.js"
            rel="nofollow"
          >
            https://codesandbox.io/s/goofy-stonebraker-pf0wg7?file=/src/index.js
          </a>
           (
          <a
            href="https://pf0wg7.csb.app/"
            rel="nofollow"
          >
            https://pf0wg7.csb.app/
          </a>
          )
        </p>
        

        <p
          dir="auto"
        >
          And there is screenshot with two requests:
        </p>
        

        <p
          dir="auto"
        >
          <a
            href="https://user-images.githubusercontent.com/1208639/189645667-bd18813b-8f6d-4987-b3af-621298bd1aff.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/1208639/189645667-bd18813b-8f6d-4987-b3af-621298bd1aff.png"
              style="max-width: 100%;"
              width="1730"
            />
          </a>
        </p>
        

        <p
          dir="auto"
        >
          What i'm doing wrong?
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/MiguelMachado-dev"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/29252011?u=ababd38859887c5e1854f1c48dfc94787ae91724&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/MiguelMachado-dev"
        >
          MiguelMachado-dev
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          Not sure, it does not happen to me. We should then wait for more people to see this. It only happen to me when using StrictMode.
        </p>
        

        <p
          dir="auto"
        >
          Edge:
          <br />
          

          <a
            href="https://user-images.githubusercontent.com/29252011/189654466-3c90adf7-6b5f-4ff7-9655-61be91455a63.png"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/29252011/189654466-3c90adf7-6b5f-4ff7-9655-61be91455a63.png"
              style="max-width: 100%;"
              width="959"
            />
          </a>
        </p>
        

        <p
          dir="auto"
        >
          Firefox:
          <br />
          

          <a
            href="https://user-images.githubusercontent.com/29252011/189654925-a59a5c23-685f-4aca-a9c7-7d278a569de6.png"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/29252011/189654925-a59a5c23-685f-4aca-a9c7-7d278a569de6.png"
              style="max-width: 100%;"
              width="1920"
            />
          </a>
        </p>
        

        <p
          dir="auto"
        >
          Both browsers up to date.
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/arpitj007"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/60127884?v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/arpitj007"
        >
          arpitj007
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          It really is an issue. Where can we find the root cause of this problem?
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/arpitj007"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/60127884?v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/arpitj007"
        >
          arpitj007
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          <a
            class="user-mention notranslate"
            data-hovercard-type="user"
            data-hovercard-url="/users/MiguelMachado-dev/hovercard"
            data-octo-click="hovercard-link-click"
            data-octo-dimensions="link_type:self"
            href="https://github.com/MiguelMachado-dev"
          >
            @MiguelMachado-dev
          </a>
           How can I contribute to fixing this bug?
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/MiguelMachado-dev"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/29252011?u=ababd38859887c5e1854f1c48dfc94787ae91724&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/MiguelMachado-dev"
        >
          MiguelMachado-dev
        </a>
        <span>
           commented 
          5 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <blockquote>
          

          <p
            dir="auto"
          >
            <a
              class="user-mention notranslate"
              data-hovercard-type="user"
              data-hovercard-url="/users/MiguelMachado-dev/hovercard"
              data-octo-click="hovercard-link-click"
              data-octo-dimensions="link_type:self"
              href="https://github.com/MiguelMachado-dev"
            >
              @MiguelMachado-dev
            </a>
             How can I contribute to fixing this bug?
          </p>
          

        </blockquote>
        

        <p
          dir="auto"
        >
          Do you know how to fix it?
        </p>
        

        <p
          dir="auto"
        >
          If you know, you can fork the project and open a PR. Or a solution, you can comment here how he can fix it properly.
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/sult4novars"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/98759712?u=5b07346681e52b12ce7007d3a35806f102305ea8&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/sult4novars"
        >
          sult4novars
        </a>
        <span>
           commented 
          4 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          Hi! I repeated this case in 
          <a
            href="https://codesandbox.io/"
            rel="nofollow"
          >
            https://codesandbox.io/
          </a>
           with clean HTML and get same result, then I try same code with corgi in 
          <a
            href="https://codepen.io/sult4novars/pen/NWMRzRX"
            rel="nofollow"
          >
            https://codepen.io/sult4novars/pen/NWMRzRX
          </a>
            and get one request)
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/sigorilla"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/1208639?u=c531dee3f3f7cdb9657a78c54bbea4081f327158&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/sigorilla"
        >
          sigorilla
        </a>
        <span>
           commented 
          4 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          Maybe it depends on OS? I have MacOS 12 (M1), Firefox 104.0.2
        </p>
        

        <p
          dir="auto"
        >
          And I reproduced it on codepen: 
          <a
            href="https://codepen.io/sult4novars/pen/NWMRzRX"
            rel="nofollow"
          >
            https://codepen.io/sult4novars/pen/NWMRzRX
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/sult4novars"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/98759712?u=5b07346681e52b12ce7007d3a35806f102305ea8&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/sult4novars"
        >
          sult4novars
        </a>
        <span>
           commented 
          4 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <blockquote>
          

          <p
            dir="auto"
          >
            Maybe it depends on OS? I have MacOS 12 (M1), Firefox 104.0.2
          </p>
          

          <p
            dir="auto"
          >
            And I reproduced it on codepen: 
            <a
              href="https://codepen.io/sult4novars/pen/NWMRzRX"
              rel="nofollow"
            >
              https://codepen.io/sult4novars/pen/NWMRzRX
            </a>
          </p>
          

        </blockquote>
        

        <p
          dir="auto"
        >
          i have same params  MacOS 12.6 (M1, 8gb ), Firefox 104.0.2
        </p>
        

        <p
          dir="auto"
        >
          <a
            href="https://user-images.githubusercontent.com/98759712/189886105-eabc296d-ccb0-43a6-b95d-e452babd031f.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/98759712/189886105-eabc296d-ccb0-43a6-b95d-e452babd031f.png"
              style="max-width: 100%;"
              width="1425"
            />
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/sult4novars"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/98759712?u=5b07346681e52b12ce7007d3a35806f102305ea8&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/sult4novars"
        >
          sult4novars
        </a>
        <span>
           commented 
          4 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <blockquote>
          

          <p
            dir="auto"
          >
            Maybe it depends on OS? I have MacOS 12 (M1), Firefox 104.0.2
          </p>
          

          <p
            dir="auto"
          >
            And I reproduced it on codepen: 
            <a
              href="https://codepen.io/sult4novars/pen/NWMRzRX"
              rel="nofollow"
            >
              https://codepen.io/sult4novars/pen/NWMRzRX
            </a>
          </p>
          

        </blockquote>
        

        <p
          dir="auto"
        >
          ooo I reproduced to but when I reload page this case not reproduced
          <br />
          
but i check status and see NS_BINDING_ABORTED and search this status, and what I found 
          <a
            href="https://stackoverflow.com/questions/704561/ns-binding-aborted-shown-in-firefox-with-httpfox"
            rel="nofollow"
          >
            https://stackoverflow.com/questions/704561/ns-binding-aborted-shown-in-firefox-with-httpfox
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/sult4novars"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/98759712?u=5b07346681e52b12ce7007d3a35806f102305ea8&v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/sult4novars"
        >
          sult4novars
        </a>
        <span>
           commented 
          4 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <blockquote>
          

          <p
            dir="auto"
          >
            Maybe it depends on OS? I have MacOS 12 (M1), Firefox 104.0.2
          </p>
          

          <p
            dir="auto"
          >
            And I reproduced it on codepen: 
            <a
              href="https://codepen.io/sult4novars/pen/NWMRzRX"
              rel="nofollow"
            >
              https://codepen.io/sult4novars/pen/NWMRzRX
            </a>
          </p>
          

        </blockquote>
        

        <p
          dir="auto"
        >
          try with new image 
          <a
            href="https://codepen.io/sult4novars/pen/NWMRzRX"
            rel="nofollow"
          >
            https://codepen.io/sult4novars/pen/NWMRzRX
          </a>
        </p>
        

        <p
          dir="auto"
        >
          when I first reload:
          <br />
          

          <a
            href="https://user-images.githubusercontent.com/98759712/189889657-dfe86d67-76db-44a7-bf2e-8bf21944fa20.png"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/98759712/189889657-dfe86d67-76db-44a7-bf2e-8bf21944fa20.png"
              style="max-width: 100%;"
              width="1203"
            />
          </a>
        </p>
        

        <p
          dir="auto"
        >
          when i try to reload many times
          <br />
          

          <a
            href="https://user-images.githubusercontent.com/98759712/189889806-3ab35744-7cc0-461a-8abd-46b03093b8b2.png"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <img
              alt="image"
              src="https://user-images.githubusercontent.com/98759712/189889806-3ab35744-7cc0-461a-8abd-46b03093b8b2.png"
              style="max-width: 100%;"
              width="1313"
            />
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    class="sc-jqUVSM dlDNbx"
    data-testid="card"
  >
    <a
      class="sc-iBkjds dyfvRb"
      href="https://github.com/kapibaara"
    >
      <img
        alt="avatar url"
        class="sc-papXJ duTIVK sc-iqcoie iXolMx"
        src="https://avatars.githubusercontent.com/u/41840872?v=4"
      />
    </a>
    <div
      class="sc-crXcEl kMFxKK"
    >
      <div
        class="sc-dkzDqf hNRMbQ"
      >
        <a
          class="sc-iBkjds dyfvRb"
          href="https://github.com/kapibaara"
        >
          kapibaara
        </a>
        <span>
           commented 
          4 days
           ago
        </span>
      </div>
    </div>
    <div
      class="sc-kDDrLX bwsbNW"
    >
      <div
        class="sc-evZas fkhGIo"
      >
        <p
          dir="auto"
        >
          when i try my example with react 17, there are not two requests in firefox
          <br />
          

          <a
            href="https://codesandbox.io/s/wizardly-lena-yobcgz?file=/src/index.js"
            rel="nofollow"
          >
            https://codesandbox.io/s/wizardly-lena-yobcgz?file=/src/index.js
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
